### ISSUE TITLE
[react-router-dom] - Export History type

### ISSUE THREAD

**Issue by cbdeveloper** [DefinitelyTyped/DefinitelyTyped#50526]
Is something wrong with CI or publishing infrastructure?

- Check the [Infrastructure status updates](https://github.com/DefinitelyTyped/DefinitelyTyped/issues/44317) thread for known issues.
- Leave a message on the [DefinitelyTyped Gitter](http://gitter.im/DefinitelyTyped/DefinitelyTyped).

If you know how to fix the issue, make a pull request instead.

- [X] I tried using the `@types/react-router-dom` package and had problems.
- [X] I tried using the latest stable version of tsc. https://www.npmjs.com/package/typescript
- [X] I understand that `@types` package versions [do not follow SemVer](https://github.com/DefinitelyTyped/DefinitelyTyped#how-do-definitely-typed-package-versions-relate-to-versions-of-the-corresponding-library).
- [X] I have a question that is inappropriate for [StackOverflow](https://stackoverflow.com/).  (Please ask any appropriate questions there).
- [X] [Mention](https://github.com/blog/821-mention-somebody-they-re-notified) the authors (see `Definitions by:` in `index.d.ts`) so they can respond.
  - Authors: @huy-nguyen @p-jackson @johnnyreilly @eps1lon @danielnixon @ynotdraw 

If you do not mention the authors the issue will be ignored.

I'm using `react-router-dom` and I need to pass the `history` object to some functions (action thunks and some other helper functions). The `@types/react-router-dom` does not export a type for the `History` object. Right now, this is how I'm handling this:

```
import type { useHistory } from "react-router-dom";

const someThunk = (history: ReturnType<typeof useHistory>) : void => {};
```

Is it possible for the `@types/react-router-dom` package to export the `History` type so consumers will be able to use it as a type?

Something like:

**@types/react-router-dom/index.d.ts**

```
import { match } from 'react-router';
import * as React from 'react';
import * as H from 'history';
export { History } from 'history';   // ADD THIS LINE ?

export {
    generatePath,
    Prompt,
    MemoryRouter,
    RedirectProps,
    Redirect,
    RouteChildrenProps,
    RouteComponentProps,
    RouteProps,
    Route,
    Router,
    StaticRouter,
    SwitchProps,
    Switch,
    match,
    matchPath,
    withRouter,
    RouterChildContext,
    useHistory,
    useLocation,
    useParams,
    useRouteMatch,
} from 'react-router';
```

Or maybe add it to `@types/react-router` and re-export it from `@types/react-router-dom`.

Is is possible or will it conflict with something?

**Comment by TrevorSundberg** [comment]
I'd love to see this exported too. A workaround in the meantime however:
```ts
type History = ReturnType<typeof useHistory>;
```

**Comment by mrcoles** [comment]
I agree these are necessary types to export. Here’s a workaround I’ve been using in my repo:

```typescript
import { RouteComponentProps } from 'react-router-dom';

/** Convenience export of the H.Location type for use with useLocation */
export type UseLocationType = RouteComponentProps['location'];

/** Convenience export of the H.History type for use with useHistory */
export type UseHistoryType = RouteComponentProps['history'];
```

IDK if better to just name them `Location` and `History` instead?

**Comment by ptmkenny** [comment]
Yes, it would be nice to have `history` exported from react-router-dom, but in the mean time you can import it from `history` package, which will already be installed if you have react-router and its types installed:

`import type { History } from 'history';`
